Create assignment crud

JudySeyram opened the pull request:
## What does this PR do ##
- This PR creates CRUD endpoints for assignment

### Description of tasks to be completed? ###
It makes sure that the following routes are working:
GET /api/assignment/home
GET /api/assignment/
GET /api/assignment/id
POST /api/assignment
PUT /api/assignment/id
DELETE /api/assignment/id

### How should this be manually tested/checked? ###
- Clone the PR
- Checkout to the branch `create-assignment-CRUD`
- Install the dependencies using `npm install`
- Start the server by running `npm start`
- Create a `.env file` and define the following as follows:
   - `PORT = <3000>`
   - `database = <database name>`
   - ` username = <your username>`
   - ` host = <127.0.0.1>`
    - ` dialect = <postgres>`
    
   - To test this feature go to http://localhost:3000/ `<test any of the routes>` on your browser.

JudySeyram committed:
Merge branch 'create-crud-for-students' of https://github.com/devcareer/class-manager into create-assignment-CRUD

talktonok committed:
Update models to use UUIDs

talktonok committed:
Merge branch 'development' of https://github.com/devcareer/class-manager into development

talktonok committed:
create seed files

talktonok committed:
create seed files

JudySeyram committed:
chore(assignment crud):add crud endpoints for assignments

talktonok committed:
create seeders
to run the seeders
make sure you run the migrations: sequelize db:migrate
then run the seeders: sequelize db:seed

TijanAyo reviewed (changes requested):
Kindly put all the information  in the config.js file into an .env file, you should not be showing details on your database out in the public.

JudySeyram committed:
feat(README.md restructure):restructure the README.md file

Hazeem01 reviewed (approved):
Nice job Judith !!
Our README now has a better and more descriptive look. 🤗

JudySeyram commented:
🤗🤗🤗 Thank you Hazeem

On Mon, 28 Nov 2022, 17:58 Hazeem01 ***@***.***> wrote:

> ***@***.**** approved this pull request.
>
> Nice job Judith !!
> Our README now has a better and more descriptive look. 🤗
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/devcareer/class-manager/pull/10#pullrequestreview-1196307017>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQFQSXPSQNPU5SLHFVYDJATWKTXFBANCNFSM6AAAAAASKLDKYQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

Ekemiben reviewed (changes requested):
Check line 7:  make sure to have Git, Node.js, and "PostgreSQL" installed on your local machine as this will enable the user to write migration commands

JudySeyram commented:
Okay. I'll do that. Thanks Ben

On Tue, 29 Nov 2022, 04:20 Ekemini Sunday Ben ***@***.***>
wrote:

> ***@***.**** requested changes on this pull request.
>
> Check line 7: make sure to have Git, Node.js, and "PostgreSQL" installed
> on your local machine as this will enable the user to write migration
> commands
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/devcareer/class-manager/pull/10#pullrequestreview-1196866861>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQFQSXMBZX6S6N4CRWNJL5DWKWABFANCNFSM6AAAAAASKLDKYQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

bellogo committed:
Merge pull request #7 from devcareer/create-crud-for-students

bellogo committed:
resolve conflicts

talktonok committed:
Merge branch 'development' of https://github.com/devcareer/class-manager into seed-database

talktonok committed:
update class student

talktonok committed:
seed database
feat(seed):seed database tables

create seeders to populate all database tables

-to run the seeders 'run npm migrate' to populate all the tables

JudySeyram committed:
edit README.md

JudySeyram committed:
Merge branch 'create-assignment-CRUD' of https://github.com/devcareer/class-manager into create-assignment-CRUD

JudySeyram committed:
initial commit

talktonok committed:
install passport

talktonok committed:
install passport

talktonok committed:
user registration and login

talktonok committed:
user registration and login

Hazeem01 committed:
chore(upload assignment):add route for student assignment upload

talktonok committed:
email verification

JudySeyram committed:
testing

Ekemiben commented on src/controllers/scoreController.js:
```diff
@@ -1,12 +1,12 @@
-import ScoreService from '../services/ScoreService.js';
```
Remove one line here since they doing the same thing

Ekemiben reviewed (approved):
Kindly remove the codes duplicate on line 1 thank you

JudySeyram committed:
delete score files

JudySeyram committed:
correction

JudySeyram committed:
correction

talktonok committed:
Authenticate verify users

JudySeyram committed:
updating branch

JudySeyram committed:
Merge branch 'feat-authentication' of https://github.com/devcareer/class-manager into create-assignment-CRUD

talktonok committed:
update

JudySeyram committed:
Merge branch 'feat-authentication' of https://github.com/devcareer/class-manager into create-assignment-CRUD

JudySeyram committed:
assignment update

talktonok committed:
fix error

talktonok committed:
fix error

talktonok committed:
Merge pull
